app/models/cart.rb

```diff
@@ -0,0 +1,20 @@
+class Cart
+  attr_reader :contents
+
+  def initialize(initial_contents)
+    @contents = initial_contents || {}
+  end
+
+  def add_freelancer(id)
+    contents[id.to_s] = (contents[id.to_s] || 0) + 1
+  end
+
+  def count_of(id)
+    contents[id.to_s]
```

Comment:
I think a nil value here would cause some problems, so as a protection/guard statement, we could do ```contents[id.to_s].to_i``` or, my preference, ```contents[id.to_s] || 0```.